gui/default/syncthing/core/syncthingController.js

```diff
@@ -563,9 +578,53 @@ angular.module('syncthing.core')
                 }
                 $scope.connections = data;
                 console.log("refreshConnections", data);
+
+                refreshRemoteGUI(data);
             }).error($scope.emitHTTPError);
         }
 
+        function refreshRemoteGUI(connections) {
+            if (!$scope.showRemoteGUI) {
+                $scope.idToRemoteGUI = {}
+                return
+            }
+            var newCache = {};
+            for (var id in connections) {
+                if (!connections.hasOwnProperty(id)) {
+                    continue;
+                }
```

Comment:
The check is redundant when iterating. Or does `hasOwnProperty` some additional magic?

Reply:
the `for ... in` loop will include inherited properties as well, whereas here only own properties are of interest.  I don't know about what inherited properties might be for the HTTP response data.  Could be avoided by using the `Object.keys(connections)` function in the loop, or for newer ECMAScript (2017 I believe) also straight `Object.entries(obj)`.

When we check if the property name (key) being a valid device ID just below, that is certainly redundant though, so removing this instance.